xkcd

```diff
@@ -26,13 +29,28 @@ function parse() {
 	# we need to decode HTML entities e.g &quot; to display text correctly
 	title="$(grep 'img src.* title=".*' "$temp_file" | cut -d '"' -f 4 | perl -MHTML::Entities -pe 'decode_entities($_);')"
 
-	rm "${temp_file}"
+	rm "$temp_file"
 }
 
 function show(){
 	parse
 	echo ""
-	kitty +kitten icat --align left "$img"
+
+	if [ $scale -eq 100 ]; then
+		kitty +kitten icat --align left "$img"
+	else
+		temp_file=$(mktemp)
+		convert "$img" -resize $scale% $temp_file
```

Comment:
For the value of the scaling, my preference would be to do a similar thing to a scale transformation in CSS and use a number with 2 decimal places e.g 1.5 for 150% bigger. That feels more intuitive to me. 

If you want to stick with a percentage value, I would include the percentage symbol like ImageMagick does.